spec/services/vendor_api/generate_test_applications_for_provider_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'rails_helper'
+
+RSpec.describe VendorAPI::GenerateTestApplicationsForProvider, sidekiq: true do
+  describe '#call' do
+    it 'generates applications to courses the provider runs' do
+      provider = create(:provider)
+      create(:course_option)
+      # rubocop:disable FactoryBot/CreateList
+      3.times do
+        create(:course_option, course: create(:course, :open_on_apply, provider: provider))
+      end
+      # rubocop:enable FactoryBot/CreateList
+
+      described_class.new.call(
+        provider: provider,
+        courses_per_application: 3,
+        count: 2,
+      )
+
+      choices = provider.application_choices
+      expect(choices.map(&:application_form).uniq.count).to eq(2)
+      expect(choices.map(&:course).map(&:provider).uniq).to eq([provider])
+    end
+
+    it 'can generate applications to courses the provider ratifies' do
+      provider = create(:provider)
+      create(:course_option)
+      create(:course_option, course: create(:course, provider: provider))
+      # rubocop:disable FactoryBot/CreateList
+      3.times do
+        create(:course_option, course: create(:course, accredited_provider: provider))
+      end
+      # rubocop:enable FactoryBot/CreateList
```

Comment:
Why not use `create_list`?

Reply:
With `create_list`, it will create three course options with the same course associated. It doesn't look there's a way using `create_list` to set a different course for each of the options (at least not one that is clearer than this)